x-pack/test/functional/services/ml/stack_management_jobs.ts

```diff
@@ -274,23 +285,116 @@ export function MachineLearningStackManagementJobsProvider(
     },
 
     async importJobs() {
-      await retry.tryForTime(5000, async () => {
-        await testSubjects.click('mlJobMgmtImportImportButton', 1000);
+      await testSubjects.click('mlJobMgmtImportImportButton', 1000);
+      await testSubjects.missingOrFail('mlJobMgmtImportJobsFlyout', { timeout: 60 * 1000 });
+    },
+
+    async assertReadErrorCalloutExists() {
+      await testSubjects.existOrFail('~mlJobMgmtImportJobsFileReadErrorCallout');
+    },
+
+    async selectExportJobType(jobType: JobType) {
+      if (jobType === 'anomaly-detector') {
+        await testSubjects.click('mlJobMgmtExportJobsADTab');
+        await testSubjects.existOrFail('mlJobMgmtExportJobsADJobList');
+      } else {
+        await testSubjects.click('mlJobMgmtExportJobsDFATab');
+        await testSubjects.existOrFail('mlJobMgmtExportJobsDFAJobList');
+      }
+    },
+
+    async selectExportJobSelectAll() {
+      await testSubjects.click('mlJobMgmtExportJobsSelectAllButton');
+    },
+
+    async getDownload(filePath: string) {
+      return retry.tryForTime(5000, async () => {
+        expect(fs.existsSync(filePath)).to.be(true);
+        return fs.readFileSync(filePath).toString();
       });
-      await retry.tryForTime(40000, async () => {
-        await testSubjects.missingOrFail('mlJobMgmtImportJobsFlyout');
+    },
+
+    getExportedFile(fileName: string) {
+      return path.resolve(REPO_ROOT, `target/functional-tests/downloads/${fileName}.json`);
+    },
+
+    deleteExportedFiles(fileNames: string[]) {
+      fileNames.forEach((file) => {
+        try {
+          fs.unlinkSync(this.getExportedFile(file));
+        } catch (e) {
+          // it might not have been there to begin with
+        }
       });
     },
 
-    async assertImportedJobIdsExist(expectedJobIds: string[]) {
-      await retry.tryForTime(40000, async () => {
-        await testSubjects.click('mlRefreshJobListButton', 1000);
-        await testSubjects.missingOrFail('mlJobMgmtImportJobsFlyout');
+    async selectExportJobs() {
+      await testSubjects.click('mlJobMgmtExportExportButton');
+    },
+
+    async assertExportedADJobsAreCorrect(expectedJobs: Array<{ job: Job; datafeed: Datafeed }>) {
+      const file = JSON.parse(
+        await this.getDownload(this.getExportedFile('anomaly_detection_jobs'))
+      );
+      const loadedFile = Array.isArray(file) ? file : [file];
+      const sortedActualJobs = loadedFile.sort((a, b) =>
+        a.job.job_id.localeCompare((a = b.job.job_id))
+      );
+
+      const sortedExpectedJobs = expectedJobs.sort((a, b) =>
+        a.job.job_id.localeCompare(b.job.job_id)
+      );
+      expect(sortedActualJobs.length).to.eql(
+        sortedExpectedJobs.length,
+        `Expected length of exported jobs to be '${sortedExpectedJobs.length}' (got '${sortedActualJobs.length}')`
+      );
+
+      sortedExpectedJobs.forEach((expectedJob, i) => {
+        expect(expectedJob.job.job_id).to.eql(
```

Comment:
We have a few cases here where we use `expect(the-expected-data).to.eql(the-actual-data)`, while it should be the other way around: `expect(the-actual-data).to.eql(the-expected-data)`.

Reply:
Updated in https://github.com/elastic/kibana/pull/110578/commits/62e901212a167e8db5384455296567e8b13963a8